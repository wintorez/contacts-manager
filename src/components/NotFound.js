import React from 'react';
import PropTypes from 'prop-types';
import Page from './Page';
import { Segment, Header } from 'semantic-ui-react';

const NotFound = ({ location }) => (
  <Page title="Page Not Found" location={location} hideMenu>
    <Segment>
      <Header as="h3">Page Not Found</Header>
    </Segment>
  </Page>
);

NotFound.propTypes = {
  location: PropTypes.object.isRequired
};

export default NotFound;
