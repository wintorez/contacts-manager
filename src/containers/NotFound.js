import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import NotFound from '../components/NotFound';

const mapStateToProps = (state, ownProps) => ({});

const mapDispatchToProps = (dispatch, ownProps) => ({});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(NotFound)
);
