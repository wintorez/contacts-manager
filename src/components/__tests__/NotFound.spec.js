import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import NotFound from '../NotFound';

const defaults = {
  location: { pathname: '/invalid-path' }
};

describe('<NotFound />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <NotFound {...defaults} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
