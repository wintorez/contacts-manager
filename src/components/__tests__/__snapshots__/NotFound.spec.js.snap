// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> renders correctly 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui container"
  >
    <div
      className="ui segment"
    >
      <h3
        className="ui header"
      >
        Page Not Found
      </h3>
    </div>
  </div>
</div>
`;
